### PR TITLE
LPD-25353 Commerce upgrade process does not take into account ctCollectionId when upgrading

### PR DESCRIPTION
**Pull request by ctang3**
https://liferay.atlassian.net/browse/LPD-25353

### CODE REVIEW

**Comment by ctang3** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java [hunk 1]
```diff
@@ -66,7 +67,7 @@ protected void doUpgrade() throws Exception {
 						_classNameLocalService.getClassNameId(
 							AccountRole.class),
 						",  classPK = ?, type_ = ", RoleConstants.TYPE_ACCOUNT,
-						" where roleId = ?"))) {
```
Comment (david-truong) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java:69]: can you alphabetize this

**Comment by ctang3** [comment]
ci:test:upgrade

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java [hunk 2]
```diff
@@ -52,9 +52,10 @@ public CommerceAccountRoleUpgradeProcess(
 	protected void doUpgrade() throws Exception {
 		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
 				StringBundler.concat(
-					"select distinct UserGroupRole.roleId from UserGroupRole ",
-					"inner join Role_ on Role_.roleId = UserGroupRole.roleId ",
-					"inner join Group_ on Group_.classNameId = '",
+					"select distinct ctCollectionId, UserGroupRole.roleId ",
```
Comment (david-truong) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java:55]: Can you update it to select by UserGroupRole.ctCollectionId

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java [hunk 3]
```diff
@@ -174,7 +178,7 @@ else if ((resourcePermission.getScope() ==
 	private boolean _hasNonaccountEntryGroup(long roleId) throws Exception {
 		try (PreparedStatement preparedStatement = connection.prepareStatement(
 				StringBundler.concat(
-					"select count(*) from (select distinct ",
+					"select count(*) from (select distinct ctCollectionId, ",
```
Comment (david-truong) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java:181]: Same here

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v8_9_1/CommerceChannelAccountEntryRelUpgradeProcess.java [hunk 4]
```diff
@@ -25,9 +25,10 @@ public class CommerceChannelAccountEntryRelUpgradeProcess
 	protected void doUpgrade() throws Exception {
 		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
 				StringBundler.concat(
-					"select CChannelAccountEntryRel.CChannelAccountEntryRelId ",
-					"from CChannelAccountEntryRel inner join ",
-					"CommercePriceList on CChannelAccountEntryRel.classPK = ",
+					"select ctCollectionId, ",
```
Comment (david-truong) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v8_9_1/CommerceChannelAccountEntryRelUpgradeProcess.java:28]: Same here

**Comment by david-truong** [comment]
@ctang3 if there is a join between two tables and they both contain ctCollectionId you should be specific with the table you want to select from.

**Comment by ctang3** [comment]
@david-truong Ah I see! Pushed my updated changes that include that.

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java [hunk 5]
```diff
@@ -175,8 +179,8 @@ private boolean _hasNonaccountEntryGroup(long roleId) throws Exception {
 		try (PreparedStatement preparedStatement = connection.prepareStatement(
 				StringBundler.concat(
 					"select count(*) from (select distinct ",
-					"UserGroupRole.groupId from UserGroupRole inner join ",
-					"Group_ on Group_.classNameId != ",
+					"UserGroupRole.ctCollectionId, UserGroupRole.groupId from ",
```
Comment (david-truong) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_1_2/CommerceAccountRoleUpgradeProcess.java:182]: Actually does this query need the ctCollectionId?

**Comment by ctang3** [comment]
Branch updated to leave the select count statement alone

**Comment by david-truong** [comment]
closing, please resend with the test

**Pull request closed without merging**